interview-questions.md

```diff
@@ -9,59 +9,66 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: This refers to the global object. when this is in strict mode certain rules apply. this can be used in a global context, function context and a class context.
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is used to build web pages by and mobile applications and react native. React is used to build user interface components, web pages and support for languages.
 
-  Researched answer:
+  Researched answer:React is used to build UI components and UI library this is used to build web pages . React is a JavaScript library that was designed by Facebook. 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: render is the only required method of  a React component lifecycle method.
 
-  Researched answer:
+  Researched answer: The render method is required and will always be called. all other methods, are called when defined. 
 
 
 
 4. What is JSX? What is one syntax difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX allows us to write HTML in React. Jsx is an extension for React using the javascript language. HTML compared to JSX has attributes while JSX has props. HTML tags have an opening and closing. JSX can be written as self closing. The nested jsx returns a single element that is called parent element. With out this wrap JSX does not compile.
 
   Researched answer:
-
+JSX stands for javascriptXML and is an extension to React and allows React to have an HTML. JSX allows for several HTML components to make a more complex html page. 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package manager that was created by facebook. this was meant to replace the npm instead is another package manager option. 
 
-  Researched answer:
+  Researched answer: yarn is a package manager for node.js, that has many features meant to replace npm, yarn includes additional features that npm could not support. In comparison Yarn can install packages from local cache, binds package versions, methods for data integrity, and multiple install packaging.
 
 
 
 6. STRETCH: What is an anonymous function in JavaScript?
 
-  Your answer:
+  Your answer: Anonymous function is a function that has no name. Anonymous functions invoke the function soon after initialization. So its for self declaring simplicity.
+  
+  Researched answer: Are functions with out a name. Since javascript supports high order functions as a result anonymous functions can be passed as a parameter into another function. 
 
-  Researched answer:
+  (function () {
+   console.log("hello");
+  })();
```

Comment:
good work! Much better!  I encourage you to really make your answers conversational.  Also, the response for #1 doesn't really say what `this` does of how it's used.  Again, I know you put time into these, but really show us your knowledge.  Remember, you're going to have to do this in interviews, so the more you can practice now, the better off you'll be down the line.